### PR TITLE
PLT-5050 (Server): API to update channel member roles.

### PR DESCRIPTION
**Pull request by grundleborg**
#### Summary
PLT-5050 (Server). API to update channel member roles.

Implements API reference proposal https://github.com/mattermost/mattermost-api-reference/pull/66

Javscript driver coming in separate WebApp PR.

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-5050

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [x] Added API documentation (required for all new APIs)
- [x] All new/modified APIs include changes to the drivers

### CODE REVIEW

**Comment by jwilander** [comment]
You've got a gofmt failure
```
Running GOFMT
diff api/channel.go gofmt/api/channel.go --- /tmp/gofmt874328822	2017-01-11 19:06:43.525358375 +0000 +++ /tmp/gofmt326563293	2017-01-11 19:06:43.525358375 +0000 @@ -1366,7 +1366,7 @@  		return  	}   -	InvalidateCacheForUser(userId); +	InvalidateCacheForUser(userId)    	rdata := map[string]string{}  	rdata[status] = ok diff api/channel_test.go gofmt/api/channel_test.go --- /tmp/gofmt926737816	2017-01-11 19:06:43.545358375 +0000 +++ /tmp/gofmt031584791	2017-01-11 19:06:43.545358375 +0000 @@ -1932,9 +1932,9 @@  	// User 1 creates a channel, making them channel admin by default.  	createChannel := model.Channel{  		DisplayName: Test API Name, -		Name: a + model.NewId() + a, -		Type: model.CHANNEL_OPEN, -		TeamId: th.BasicTeam.Id, +		Name:        a + model.NewId() + a, +		Type:        model.CHANNEL_OPEN, +		TeamId:      th.BasicTeam.Id,  	}    	rchannel, err := th.BasicClient.CreateChannel(&createChannel)
/bin/sh: 1: [: diff api/channel.go gofmt/api/channel.go --- /tmp/gofmt874328822	2017-01-11 19:06:43.525358375 +0000 +++ /tmp/gofmt326563293	2017-01-11 19:06:43.525358375 +0000 @@ -1366,7 +1366,7 @@  		return  	}   -	InvalidateCacheForUser(userId); +	InvalidateCacheForUser(userId)    	rdata := map[string]string{}  	rdata[status] = ok diff api/channel_test.go gofmt/api/channel_test.go --- /tmp/gofmt926737816	2017-01-11 19:06:43.545358375 +0000 +++ /tmp/gofmt031584791	2017-01-11 19:06:43.545358375 +0000 @@ -1932,9 +1932,9 @@  	// User 1 creates a channel, making them channel admin by default.  	createChannel := model.Channel{  		DisplayName: Test: unexpected operator
gofmt failure
```

**Comment by grundleborg** [comment]
Oops. I always forget that `make test` doesn't include running `gofmt`. Fixed and ready for review again now.

File: model/client.go [hunk 1]
```diff
@@ -2334,3 +2334,17 @@ func (c *Client) ListReactions(channelId string, postId string) ([]*Reaction, *A
 		return ReactionsFromJson(r.Body), nil
 	}
 }
+
+func (c *Client) UpdateChannelRoles(channelId string, userId string, roles string) (*Result, *AppError) {
```
Comment (jwilander) [inline model/client.go]: Can you use the new method here with `ResponseMetadata`? Similar to this https://github.com/mattermost/platform/blob/master/model/client.go#L495

File: model/client.go [hunk 2]
```diff
@@ -2334,3 +2334,25 @@ func (c *Client) ListReactions(channelId string, postId string) ([]*Reaction, *A
 		return ReactionsFromJson(r.Body), nil
 	}
 }
+
```
Comment (jwilander) [inline model/client.go]: Sorry just one more thing, can you add a comment detailing what the function does, returns, etc.?

Comment (grundleborg) [inline model/client.go]: Done.

**Pull request merged**